lib/node/rpc.js

```diff
@@ -2346,10 +2347,99 @@ class RPC extends RPCBase {
     return result;
   }
 
+  async deriveAddresses(args, help) {
+    if (help || args.length > 2 || args.length === 0)
+      throw new RPCError(errs.MISC_ERROR, 'deriveaddresses "descriptor"');
+
+    const valid = new Validator(args);
+
+    const desc = parseDescriptor(valid.str(0, ''), this.network, true);
+
+    if (desc.isRange() && args.length === 1) {
+      throw new RPCError(
+        errs.INVALID_PARAMETER,
+        'Range must be specified for ranged descriptor'
+      );
+    }
+
+    if (!desc.isRange() && args.length > 1) {
+      throw new RPCError(
+        errs.INVALID_PARAMETER,
+        'Range should not be specified for un-ranged descriptor'
+      );
+    }
+
+    let low = 0, high = 0;
+
+    try {
+      high = valid.u32(1, 0);
+    } catch (e) {
+      try {
+        const arr = valid.array(1, []);
+        low = arr[0];
+        high = arr[1];
+      } catch (e) {
+        throw new RPCError(
+          errs.INVALID_PARAMETER,
+          'Invalid range.'
+        );
+      }
+    }
+
+    this.isValidRange(low, high);
+
+    const addresses = [];
+
+    for (let i = low; i <= high; i++) {
+      const address = desc.getAddresses(i);
+
+      for (let j = 0; j < address.length; j++) {
+        addresses.push(address[j]);
+      };
+    }
+
+    return addresses;
+  }
+
   /*
    * Helpers
    */
 
+  isValidRange(low, high) {
+    if (!Number.isInteger(low)) {
+      throw new RPCError(
+        errs.INVALID_PARAMETER, 'Range begin must be an integer'
+      );
+    }
+
+    if (!Number.isInteger(high)) {
+      throw new RPCError(
+        errs.INVALID_PARAMETER, 'Range end must be an integer'
+      );
+    }
```

Comment:
I believe this check is redundant. `Validator` class will ensure it is a `u32int`

Reply:
This check is to ensure the entries in range array are valid. I think its required there.